lua/entities/gmod_wire_spu/cl_spuvm.lua

```diff
@@ -211,8 +209,83 @@ function VM:ReadString(address)
   return charString
 end
 
+--------------------------------------------------------------------------------
+-- Calculate ADSR Envelope
+--------------------------------------------------------------------------------
 
-
+function VM:CalculateADSR(deltaTime)
+  for chan, _ in pairs(self.Channel) do
```

Comment:
why not `for _, chan in pairs(self.Channel) do` and replace all `self.Channel[chan]` with `chan`?

Reply:
Good point, I'll fix that real quick.